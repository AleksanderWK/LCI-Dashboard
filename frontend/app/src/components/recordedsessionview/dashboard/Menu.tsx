import React from "react";
import {
    createStyles,
    ListItemIcon,
    makeStyles,
    Menu as MUIMenu,
    MenuItem,
    MenuList,
    Theme,
    Typography
} from "@material-ui/core";
import DeleteOutlinedIcon from "@material-ui/icons/DeleteOutlined";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        menuList: {
            padding: 0,
            color: theme.palette.text.default,
            "& .MuiListItemIcon-root": {
                minWidth: 0,
                paddingRight: theme.spacing(1)
            }
        }
    })
);

interface Props {
    open: boolean;
    anchorEl: HTMLElement | null;
    onRemoveView: () => void;
    onMenuClose: () => void;
}

export default function Menu(props: Props): JSX.Element {
    const classes = useStyles();

    return (
        <MUIMenu
            className={"noDrag"}
            anchorEl={props.anchorEl}
            getContentAnchorEl={null}
            disableAutoFocusItem={true}
            disableScrollLock={true}
            keepMounted
            open={props.open}
            onClose={props.onMenuClose}
            anchorOrigin={{
                vertical: "bottom",
                horizontal: "right"
            }}
            transformOrigin={{
                vertical: "top",
                horizontal: "right"
            }}
            elevation={3}
            data-testid="dropdown-menu"
        >
            <MenuList dense={true} className={classes.menuList}>
                <MenuItem onClick={props.onRemoveView} aria-label="remove view">
                    <ListItemIcon>
                        <DeleteOutlinedIcon />
                    </ListItemIcon>
                    <Typography variant="inherit">Remove view</Typography>
                </MenuItem>
            </MenuList>
        </MUIMenu>
    );
}
