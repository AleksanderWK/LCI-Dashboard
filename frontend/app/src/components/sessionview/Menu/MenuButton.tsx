import {createStyles, makeStyles, Theme, Tooltip} from "@material-ui/core";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        buttonWrapper: {
            position: "relative",
            display: "flex",
            justifyContent: "center"
        },
        button: {
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            width: "60px",
            height: "60px",
            borderRadius: "50%",
            border: "3px solid",
            borderColor: theme.palette.primary.main,
            boxShadow: "0px 2px 5px 0px rgba(0,0,0,0.25)",
            outline: 0,
            cursor: "pointer",
            padding: 0,
            margin: 0,
            color: theme.palette.text.default,
            transition: "box-shadow 0.15s linear",
            "&:hover": {
                boxShadow: "0px 2px 5px 0px rgba(0,0,0,0.4)"
            }
        },
        tooltip: {
            marginLeft: 10
        },
        indicator: {
            position: "absolute",
            bottom: 7,
            right: 0,
            backgroundColor: "#D6D8DD",
            width: "6px",
            height: "46px",
            borderRadius: "10px 0 0 10px"
        }
    })
);

interface Props {
    children: JSX.Element;
    selected: boolean;
    tooltip: string;
    onClick: () => void;
}

export default function MenuButton(props: Props): JSX.Element {
    const classes = useStyles();

    return (
        <div className={classes.buttonWrapper}>
            <Tooltip title={props.tooltip} classes={{tooltipPlacementRight: classes.tooltip}} placement="right">
                <button className={classes.button} onClick={props.onClick}>
                    {props.children}
                </button>
            </Tooltip>
            {props.selected && <div className={classes.indicator}></div>}
        </div>
    );
}
