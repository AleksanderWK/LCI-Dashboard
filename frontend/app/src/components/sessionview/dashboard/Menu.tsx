import React from "react";
import {
    createStyles,
    ListItemIcon,
    makeStyles,
    Menu as MUIMenu,
    MenuItem,
    MenuList,
    Theme,
    Typography
} from "@material-ui/core";
import ZoomInIcon from "@material-ui/icons/ZoomIn";
import ZoomOutIcon from "@material-ui/icons/ZoomOut";
import DeleteOutlinedIcon from "@material-ui/icons/DeleteOutlined";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        menuList: {
            padding: 0,
            color: theme.palette.text.default,
            "& .MuiListItemIcon-root": {
                minWidth: 0,
                paddingRight: theme.spacing(1)
            }
        }
    })
);

interface Props {
    open: boolean;
    anchorEl: HTMLElement | null;
    isDetailedView: boolean;
    onShowMore: () => void;
    onShowLess: () => void;
    onRemoveView: () => void;
    onMenuClose: () => void;
}

/**
 * A dropdown menu for a container in the session view.
 * Has a buttons to switch type of display and remove container.
 * @param {object} props - Component props
 * @param {boolean} props.open - Whether the dropdown is open or not
 * @param {HTMLElement | null} props.anchorEl - The anchor element for the dropdown menu
 * @param {() => void} props.onShowMore - A callback function that is called when show more button is clicked
 * @param {() => void} props.onShowLess - A callback function that is called when show less button is clicked
 * @param {() => void} props.onRemoveView - A callback function that is called when remove view button is clicked
 * @param {() => void} props.onMenuClose - A callback function that is called dropdown menu is closed
 */
export default function Menu(props: Props): JSX.Element {
    const classes = useStyles();

    return (
        <MUIMenu
            className={"noDrag"}
            anchorEl={props.anchorEl}
            getContentAnchorEl={null}
            disableAutoFocusItem={true}
            disableScrollLock={true}
            keepMounted
            open={props.open}
            onClose={props.onMenuClose}
            anchorOrigin={{
                vertical: "bottom",
                horizontal: "right"
            }}
            transformOrigin={{
                vertical: "top",
                horizontal: "right"
            }}
            elevation={3}
            data-testid="dropdown-menu"
        >
            <MenuList dense={true} className={classes.menuList}>
                {props.isDetailedView ? (
                    <MenuItem onClick={props.onShowLess} aria-label="show less">
                        <ListItemIcon>
                            <ZoomOutIcon />
                        </ListItemIcon>
                        <Typography variant="inherit">Show less</Typography>
                    </MenuItem>
                ) : (
                    <MenuItem onClick={props.onShowMore} aria-label="show more">
                        <ListItemIcon>
                            <ZoomInIcon />
                        </ListItemIcon>
                        <Typography variant="inherit">Show more</Typography>
                    </MenuItem>
                )}

                <MenuItem onClick={props.onRemoveView} aria-label="remove view">
                    <ListItemIcon>
                        <DeleteOutlinedIcon />
                    </ListItemIcon>
                    <Typography variant="inherit">Remove view</Typography>
                </MenuItem>
            </MenuList>
        </MUIMenu>
    );
}
