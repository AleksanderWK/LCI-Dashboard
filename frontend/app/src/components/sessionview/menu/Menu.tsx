import React from "react";
import {createStyles, Fab, makeStyles, Theme, Tooltip} from "@material-ui/core";
import AddIcon from "@material-ui/icons/Add";
import ClassroomButton from "./ClassroomButton";
import StudentButton from "./StudentButton";
import {useRecoilValue, useSetRecoilState} from "recoil";
import {createSessionPopupOpenState} from "../../../state/popup";
import {selectedSessionIdState, sessionsState} from "../../../state/session";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        menu: {
            position: "relative",
            boxSizing: "border-box",
            padding: "30px 0",
            width: "100%",
            height: "100%",
            backgroundColor: "#E6E6E6", // Background of title bar
            overflow: "scroll",
            gridColumnStart: 1,
            gridColumnEnd: 2,
            gridRowStart: 1,
            gridRowEnd: 3,
            // Hide scroll bar
            msOverflowStyle: "none",
            scrollbarWidth: "none",
            "&::-webkit-scrollbar": {
                display: "none"
            }
        },
        wrapper: {
            display: "grid",
            gridTemplateColumns: "100%",
            gap: "20px"
        },
        hr: {
            margin: "0 auto",
            width: "40px",
            borderRadius: "2px",
            borderBottom: "1px solid #878787"
        },
        tooltip: {
            marginLeft: 10
        },
        addButton: {
            margin: "0 auto"
        },
        addIcon: {
            color: theme.palette.text.default
        }
    })
);

export default function Menu(): JSX.Element {
    const classes = useStyles();

    const setCreateSessionPopupOpen = useSetRecoilState(createSessionPopupOpenState);

    const selectedSessionId = useRecoilValue(selectedSessionIdState);

    const sessions = useRecoilValue(sessionsState);

    return (
        <div className={classes.menu}>
            <div className={classes.wrapper}>
                <ClassroomButton selected={!selectedSessionId} />

                <div className={classes.hr}></div>

                {sessions.map((session) => (
                    <StudentButton
                        key={session._id}
                        sessionId={session._id}
                        selected={session._id === selectedSessionId}
                        studentName={session.studentName}
                    />
                ))}
                <Tooltip
                    title="Create new session"
                    classes={{tooltipPlacementRight: classes.tooltip}}
                    placement="right"
                >
                    <Fab
                        size="small"
                        color="primary"
                        aria-label="create new learning session"
                        className={classes.addButton}
                        onClick={() => {
                            setCreateSessionPopupOpen(true);
                        }}
                    >
                        <AddIcon className={classes.addIcon} />
                    </Fab>
                </Tooltip>
            </div>
        </div>
    );
}
