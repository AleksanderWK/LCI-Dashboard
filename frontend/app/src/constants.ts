// Frequency in Hz (cycles per second)
export const FREQUENCY = 2;

// Maximum time interval to show on live charts in seconds
export const LIVE_CHART_RANGE = 30;

export enum Variable {
    CognitiveLoad = "cl",
    PerceivedDifficulty = "pd",
    Familiarity = "fam",
    InformationProcessingIndex = "ipi",
    PhysiologicalArousal = "pa",
    Engagement = "eng",
    PhysiologicalStress = "ps",
    EmotionalRegulation = "er",
    MotionStability = "ms",
    EnergySpentFatigue = "esf",
    EducationalSpecificEmotions = "ese"
}

export enum Device {
    EyeTracker = "Eye Tracker",
    Wristband = "Wristband",
    VideoBody = "Video of Body",
    VideoFace = "Video of Face"
}

export enum EyeTrackingDevice {
    Mobile = "Mobile",
    Stationary = "Stationary"
}

export interface MMDVariableInfo {
    name: string;
    device: Device;
    description: string;
    maxValue?: number;
    calculationTime?: number;
    enabled: boolean;
}

export const MMDVariables: {[key in Variable]: MMDVariableInfo} = {
    [Variable.CognitiveLoad]: {
        name: "Cognitive Load",
        device: Device.EyeTracker,
        description:
            "Cognitive load refers to the amount of information that the working memory of a person can hold at once. In an educational setting, it is the effort needed for a learner to learn new information.",
        calculationTime: 30,
        enabled: true
    },
    [Variable.PerceivedDifficulty]: {
        name: "Perceived Difficulty",
        device: Device.EyeTracker,
        description:
            "Perceived difficulty is the perception of the ease or difficulty of performing in an educational context.",
        maxValue: 100,
        enabled: true
    },
    [Variable.Familiarity]: {
        name: "Familiarity",
        device: Device.EyeTracker,
        description: "",
        enabled: false
    },
    [Variable.InformationProcessingIndex]: {
        name: "Information Processing Index",
        device: Device.EyeTracker,
        description:
            "The ratio of local (100) and global (0) information processing. Staring at one point is local processing, while rapidly scanning a larger area is global processing.",
        maxValue: 100,
        enabled: true
    },
    [Variable.PhysiologicalArousal]: {
        name: "Physiological Arousal",
        device: Device.Wristband,
        description:
            "Physiological Arousal is how strong the current emotions are. It is calculated by the average EDA in a time interval of 8 seconds.",
        calculationTime: 8,
        enabled: true
    },
    [Variable.Engagement]: {
        name: "Engagement",
        device: Device.Wristband,
        description:
            "Engagement is a measure of involvement and enthusiasm for the activity. It is calculated by the area under the graph of tonic EDA, in a time interval of 4 seconds.",
        calculationTime: 4,
        enabled: true
    },
    [Variable.PhysiologicalStress]: {
        name: "Physiological Stress",
        device: Device.Wristband,
        description:
            "Physioloigcal Stress is indicated by LF/HF, the relative strengths of the low frequency and high frequency waves of the heart pulse signal.",
        calculationTime: 300,
        enabled: true
    },
    [Variable.EmotionalRegulation]: {
        name: "Emotional Regulation",
        device: Device.Wristband,
        description: "",
        enabled: false
    },
    [Variable.MotionStability]: {
        name: "Motion Stability",
        device: Device.VideoBody,
        description: "",
        enabled: false
    },
    [Variable.EnergySpentFatigue]: {
        name: "Energy Spent/Fatigue",
        device: Device.VideoBody,
        description:
            "The amount of energy spent by the student, also an indicator of fatigue. This is determined by calculating the jerk, the third derivative of the position skeletal body.",
        calculationTime: 3,
        enabled: true
    },
    [Variable.EducationalSpecificEmotions]: {
        name: "Educational-specific Emotions",
        device: Device.VideoFace,
        description: "Emotions that the student is experiencing in an educational setting.",
        enabled: true
    }
};

export const emotionsColorMapper: {[key: string]: string} = {
    boredom: "#F15025",
    frustration: "#191919",
    confusion: "#FFE45E",
    delight: "#6782B7"
};

export const emotionsIndexMapper: {[key: string]: number} = {
    boredom: 0,
    frustration: 1,
    confusion: 2,
    delight: 3
};

export const months = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December"
];
