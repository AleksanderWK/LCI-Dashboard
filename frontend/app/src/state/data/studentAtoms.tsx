import {atom} from "recoil";

export const students = atom({
    key: "students",
    default: ["John Doe", "Jane Smith"]
});
