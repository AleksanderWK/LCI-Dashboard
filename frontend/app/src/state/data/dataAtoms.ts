import {atom, atomFamily, selector, selectorFamily} from "recoil";
import {Variable} from "../../constants";

interface Data {
    [Variable.CognitiveLoad]: [number, number][];
    [Variable.PerceivedDifficulty]: [number, number][];
    [Variable.Familiarity]: [number, number][];
    [Variable.InformationProcessingIndex]: [number, number][];
    [Variable.PhysiologicalArousal]: [number, number][];
    [Variable.Engagement]: [number, number][];
    [Variable.PhysiologicalStress]: [number, number][];
    [Variable.EmotionalRegulation]: [number, number][];
    [Variable.MotionStability]: [number, number][];
    [Variable.EnergySpentFatigue]: [number, number][];
    [Variable.EducationalSpecificEmotions]: [number, number][];
}

/*
 *  An atom that stores which student is selected
 */
const selectedStudentIdState = atom<number | null>({
    key: "selectedStudentId",
    default: null
});

/*
 *   An atomFamily that stores the data for each student
 */
const dataState = atomFamily<Data, number | null>({
    key: "data",
    default: {
        [Variable.CognitiveLoad]: [],
        [Variable.PerceivedDifficulty]: [],
        [Variable.Familiarity]: [],
        [Variable.InformationProcessingIndex]: [],
        [Variable.PhysiologicalArousal]: [],
        [Variable.Engagement]: [],
        [Variable.PhysiologicalStress]: [],
        [Variable.EmotionalRegulation]: [],
        [Variable.MotionStability]: [],
        [Variable.EnergySpentFatigue]: [],
        [Variable.EducationalSpecificEmotions]: []
    }
});

/*
 *  A selector that returns the selected student's data
 */
const selectedStudentDataState = selector<Data>({
    key: "selectedStudentData",
    get: ({get}) => {
        const id = get(selectedStudentIdState);

        return get(dataState(id));
    },
    set: ({get, set}, newValue) => {
        const id = get(selectedStudentIdState);

        set(dataState(id), newValue);
    }
});

/*
 *  A selector that returns the last value for a given variable from the selected student's data
 */
const selectedStudentLastValueState = selectorFamily<number | null, Variable>({
    key: "selectedStudentLastValue",
    get: (variable: Variable) => ({get}) => {
        const id = get(selectedStudentIdState);

        const data = get(dataState(id))[variable];

        if (data.length > 0) {
            return data.slice(-1)[0][1];
        } else {
            return null;
        }
    }
});

export {selectedStudentIdState, dataState, selectedStudentDataState, selectedStudentLastValueState};
