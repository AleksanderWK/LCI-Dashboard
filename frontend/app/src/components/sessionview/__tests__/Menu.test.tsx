import React from "react";
import ReactDOM from "react-dom";
import {render, screen} from "@testing-library/react";
import {RecoilRoot} from "recoil";
import {selectedSessionIdState, sessionIdsState, sessionState} from "../../../state/data/dataAtoms";
import {studentIdsState, studentState} from "../../../state/data/studentAtoms";
import {EyeTrackingDevice} from "../../../constants";
import Menu from "../Menu/Menu";

const testSessionIds = [1, 2];
const testStudentIds = [1, 2];

const testStudents = {1: {_id: 1, name: "John Doe"}, 2: {_id: 2, name: "Jane Smith"}};

const testSessions = {
    1: {sessionId: 1, studentId: 1, eyeTrackingDevice: EyeTrackingDevice.Mobile, recording: false},
    2: {sessionId: 2, studentId: 2, eyeTrackingDevice: EyeTrackingDevice.Stationary, recording: true}
};

const testSelectedSessionId = 1;

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <RecoilRoot
            initializeState={(snap) => {
                snap.set(sessionIdsState, testSessionIds);
                snap.set(studentIdsState, testStudentIds);
                snap.set(studentState(1), testStudents[1]);
                snap.set(studentState(2), testStudents[2]);
                snap.set(sessionState(1), testSessions[1]);
                snap.set(sessionState(2), testSessions[2]);
                snap.set(selectedSessionIdState, testSelectedSessionId);
            }}
        >
            <Menu />
        </RecoilRoot>,
        div
    );
});

it("matches snapshot", () => {
    const {baseElement} = render(
        <RecoilRoot
            initializeState={(snap) => {
                snap.set(sessionIdsState, testSessionIds);
                snap.set(studentIdsState, testStudentIds);
                snap.set(studentState(1), testStudents[1]);
                snap.set(studentState(2), testStudents[2]);
                snap.set(sessionState(1), testSessions[1]);
                snap.set(sessionState(2), testSessions[2]);
                snap.set(selectedSessionIdState, testSelectedSessionId);
            }}
        >
            <Menu />
        </RecoilRoot>
    );
    expect(baseElement).toMatchSnapshot();
});

it("renders correct based on state", () => {
    render(
        <RecoilRoot
            initializeState={(snap) => {
                snap.set(sessionIdsState, testSessionIds);
                snap.set(studentIdsState, testStudentIds);
                snap.set(studentState(1), testStudents[1]);
                snap.set(studentState(2), testStudents[2]);
                snap.set(sessionState(1), testSessions[1]);
                snap.set(sessionState(2), testSessions[2]);
                snap.set(selectedSessionIdState, testSelectedSessionId);
            }}
        >
            <Menu />
        </RecoilRoot>
    );

    expect(screen.getByText(/JD/i)).toBeInTheDocument();
    expect(screen.getByText(/JS/i)).toBeInTheDocument();
    expect(screen.getByText(/REC/i)).toBeInTheDocument();
    expect(screen.getAllByText(/REC/i)).toHaveLength(1);
});
