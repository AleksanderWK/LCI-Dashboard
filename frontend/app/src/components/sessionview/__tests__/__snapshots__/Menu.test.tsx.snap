// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-menu-1"
    >
      <div
        class="makeStyles-wrapper-2"
      >
        <div
          class="makeStyles-buttonWrapper-8"
        >
          <button
            class="makeStyles-button-9"
            title="All sessions"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-7"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <rect
                fill="none"
                height="24"
                width="24"
              />
              <g>
                <path
                  d="M12,12.75c1.63,0,3.07,0.39,4.24,0.9c1.08,0.48,1.76,1.56,1.76,2.73L18,18H6l0-1.61c0-1.18,0.68-2.26,1.76-2.73 C8.93,13.14,10.37,12.75,12,12.75z M4,13c1.1,0,2-0.9,2-2c0-1.1-0.9-2-2-2s-2,0.9-2,2C2,12.1,2.9,13,4,13z M5.13,14.1 C4.76,14.04,4.39,14,4,14c-0.99,0-1.93,0.21-2.78,0.58C0.48,14.9,0,15.62,0,16.43V18l4.5,0v-1.61C4.5,15.56,4.73,14.78,5.13,14.1z M20,13c1.1,0,2-0.9,2-2c0-1.1-0.9-2-2-2s-2,0.9-2,2C18,12.1,18.9,13,20,13z M24,16.43c0-0.81-0.48-1.53-1.22-1.85 C21.93,14.21,20.99,14,20,14c-0.39,0-0.76,0.04-1.13,0.1c0.4,0.68,0.63,1.46,0.63,2.29V18l4.5,0V16.43z M12,6c1.66,0,3,1.34,3,3 c0,1.66-1.34,3-3,3s-3-1.34-3-3C9,7.34,10.34,6,12,6z"
                />
              </g>
            </svg>
          </button>
        </div>
        <div
          class="makeStyles-hr-3"
        />
        <div
          class="makeStyles-buttonWrapper-8"
        >
          <button
            class="makeStyles-button-9"
            title="John Doe"
          >
            <div
              class="makeStyles-initialsContainer-12"
            >
              <p
                class="MuiTypography-root makeStyles-initials-13 MuiTypography-body1"
              >
                JD
              </p>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-14"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </button>
          <div
            class="makeStyles-indicator-11"
          />
        </div>
        <div
          class="makeStyles-buttonWrapper-8"
        >
          <button
            class="makeStyles-button-9"
            title="Jane Smith"
          >
            <div
              class="makeStyles-initialsContainer-12"
            >
              <p
                class="MuiTypography-root makeStyles-initials-13 MuiTypography-body1"
              >
                JS
              </p>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-14"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <div
              class="makeStyles-recordingLabelContainer-15"
            >
              <p
                class="MuiTypography-root makeStyles-recordingLabel-16 MuiTypography-body1"
              >
                REC
              </p>
            </div>
          </button>
        </div>
        <button
          aria-label="create new learning session"
          class="MuiButtonBase-root MuiFab-root makeStyles-addButton-5 MuiFab-sizeSmall MuiFab-primary"
          tabindex="0"
          title="Create new session"
          type="button"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-addIcon-6"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
